pkg/api/resources.go

```diff
@@ -13,4 +17,6 @@ type ContainerResources struct {
 	// MemoryReservation is the minimum amount of memory (in bytes) the container needs to run efficiently.
 	// TODO: implement a placement constraint that checks available memory on machines.
 	MemoryReservation int64
+	// DeviceRequests for access to things like GPUs
+	DeviceRequests []container.DeviceRequest
```

Comment:
Just a tiny nit, regardless of the name of the docker internal struct, let's call it `DeviceReservations` to be consistent with the compose terminology and config (limits and reservations) and the `MemoryReservation` above.